### PR TITLE
Refactor Extension Modules

### PR DESCRIPTION
**Pull request by WillAyd**
This puts all source files in `pandas/_libs/src` and all header files in `pandas/_libs/include` This simplifies our includes, letting us do them in an "absolute" way, while also making it clear which headers/sources were vendored

### CODE REVIEW

File: pandas/_libs/tslibs/meson.build [hunk 1]
```diff
@@ -4,38 +4,27 @@ tslibs_sources = {
...
-                   'include_dirs': inc_datetime},
-    'timezones': {'sources': ['timezones.pyx', 'src/datetime/np_datetime.c'],
-                  'include_dirs': inc_datetime},
-    'tzconversion': {'sources': ['tzconversion.pyx', 'src/datetime/np_datetime.c'],
-                     'include_dirs': inc_datetime},
-    'vectorized': {'sources': ['vectorized.pyx', 'src/datetime/np_datetime.c'],
-                    'include_dirs': inc_datetime}
+    'np_datetime': {'sources': ['np_datetime.pyx']},
+    'offsets': {'sources': ['offsets.pyx']},
+    'parsing': {'sources': ['parsing.pyx', '../src/parser/tokenizer.c']},
```
Comment (WillAyd) [inline pandas/_libs/tslibs/meson.build:12]: I'm not sure why this still needs tokenizer.c - should be replaced via the capsule import but likely missed

**Review by mroeschke — APPROVED** [review]
Nice! I like the `/vendored/` distinction

**Comment by mroeschke** [comment]
Thanks @WillAyd

**Comment by WillAyd** [comment]
Try to wipe your build folder and start from scratch with meson. I don't know that meson handles changes in the file system that gracefully compared to our old methodology

**Pull request merged**